src/CloudSearchDomain/CloudSearchDomainFactory.php

```diff
@@ -26,7 +26,7 @@ protected function addDefaultArgs(&$args)
         if (strpos($args['endpoint'], 'http') !== 0) {
             $args['endpoint'] = Url::buildUrl([
                 'host'   => $args['endpoint'],
-                'scheme' => $args['scheme'],
+                'scheme' => 'https',
```

Comment:
Hmmm... there is a potential problem, because `CloudSearchClient::describeDomains` does not include a scheme when you retrieve the endpoint. This is because the endpoints are accessible via http and https, and it is left up to the user to choose. I had the code set up the way it was, because it is likely that the endpoint provided won't include a scheme, especially when using the [`CloudSearchClient::getDomainClient` method](https://github.com/aws/aws-sdk-php/blob/v3/src/CloudSearch/CloudSearchClient.php#L21).